test_launch_ros/test/test_launch_ros/actions/test_set_remap.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2020 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for the SetRemap Action."""
+
+from launch import LaunchContext
+from launch.actions import PopLaunchConfigurations
+from launch.actions import PushLaunchConfigurations
+
+from launch_ros.actions import Node
+from launch_ros.actions import SetRemap
+from launch_ros.actions.load_composable_nodes import get_composable_node_load_request
+from launch_ros.descriptions import ComposableNode
+
+import pytest
+
+
+class MockContext:
+
+    def __init__(self):
+        self.launch_configurations = {}
+
+    def perform_substitution(self, sub):
+        return sub.perform(None)
+
+
+def get_set_remap_test_remaps():
+    return [
+        pytest.param(
+            [('from', 'to')],
+            id='One remapping rule'
+        ),
+        pytest.param(
+            [('from1', 'to1'), ('from2', 'to2')],
+            id='Two remapping rules'
+        ),
+    ]
+
+
+@pytest.mark.parametrize(
+    'remapping_rules',
+    get_set_remap_test_remaps()
+)
+def test_set_remap(remapping_rules):
+    lc = MockContext()
+    for src, dst in remapping_rules:
+        SetRemap(src, dst).execute(lc)
+    assert lc.launch_configurations == {'ros_remaps': remapping_rules}
+
+
+def test_set_remap_is_scoped():
+    lc = LaunchContext()
+    push_conf = PushLaunchConfigurations()
+    pop_conf = PopLaunchConfigurations()
+    set_remap = SetRemap('from', 'to')
+
+    push_conf.execute(lc)
+    set_remap.execute(lc)
+    assert lc.launch_configurations == {'ros_remaps': [('from', 'to')]}
+    pop_conf.execute(lc)
+    assert lc.launch_configurations == {}
+
+
+def test_set_remap_with_node():
+    lc = MockContext()
+    node = Node(
+        package='asd',
+        executable='bsd',
+        name='my_node',
+        namespace='my_ns',
+        remappings=[('from2', 'to2')]
+    )
+    set_remap = SetRemap('from1', 'to1')
```

Comment:
@ivanpauno mind to add another test to verify that global remaps take precedence?

Reply:
> @ivanpauno mind to add another test to verify that global remaps take precedence?

They take precedence by being prepended (because `rcl` precedence works like that for remapping rules).
This test case already tests correctly that the "global" remapping rule is prepended.

The test in L55 already verifies (for the second parameter), that the `SetRemap` action at the upper scope will take precedence.

Reply:
Yeah, misread it.